### ISSUE TITLE
FAC - Bug: Some reports come down with random names

### ISSUE THREAD

**Issue by jadudm** [GSA-TTS/FAC#4394]
### Describe the bug

This report:

https://app.fac.gov/dissemination/summary/2023-03-GSAFAC-0000000334

does not come down with the name

2023-03-GSAFAC-0000000334.pdf

(or whatever reports should come down named). Instead, it comes down with the name

`2022APR0001000334.pdf`

which is not our name (I don't think).

We should probably make sure all reports always come down with a consistent name based on the report id. `sar-<report-id>.pdf`, or perhaps just `<report-id>.pdf`. 

### Steps to reproduce the bug

Download that audit.

### Expected Behavior

The name should be ours, and consistent.

### Code of Conduct

- [X] I agree to abide by the [Digital.gov Community Guidelines](https://digital.gov/communities/community-guidelines/) and the [TTS Code of Conduct](https://handbook.tts.gsa.gov/code-of-conduct/). Respect your peers, use plain language, and be patient.
- [X] I checked the [current issues](https://github.com/GSA-TTS/FAC/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by danswick** [comment]
I did a little bit of repo-spelunking and think I understand how this happened. The gist is that the PDF in question was uploaded before we switched to the current report ID format and possibly before we started using the report ID as the filename. [Here's where the change to report ID format happened](https://github.com/GSA-TTS/FAC/pull/2329).

* Report was started and the PDF was uploaded on October 2nd, which, oddly, is the exact day Tadhg proposed the new report ID format. But those changes weren't deployed until the next day, October 3rd. Incredibly, this issue wouldn't exist if the PDF had been upload a single day later. 
* The PDF report ID is a year earlier than the actual report ID and is in April instead of the March. This confused me at first until I remembered that we switched the from using the fiscal year *start date* to the fiscal year *end date*. 
* The old report ID format matches the format of the PDF filename.
